### DISCUSSION TITLE
Compression of existing content?

### DISCUSSION THREAD

**Discussion by TonyNoble** [minio/minio#17816] [Q&A] [answered]
While investigating disk usage of our minio install, we've spotted that we could make significant savings by compressing certain file types that aren't covered by the default rules.

We've updated the compressions patterns to take account of this going forward, but we're still left with a large number (approx 400k) of existing files that would ideally be compressed, but aren't.  

So the question is - is there any straightforward way to apply minio's compression to the content that's already there?  An 'mc' command, or similar?

(Apologies if this has already been answered, my google-fu has failed me on this one)

**Comment by klauspost** [comment, marked as answer]
There is currently no such functionality. We have a batch operation for this planned, but no timeframe.

You can of course copy the content with mc after enabling compression for these types - that is the only currently working solution for this.

**Reply by TonyNoble** [reply]
Hi @klauspost - thanks for the speedy answer.  If I used 'mv' rather than 'cp' would that also work?  Or would it bypass the compression?  Disk space is tight enough that I don't really have room to perform a 'copy-and-delete' of the entire bucket contents...

**Reply by klauspost** [reply]
Yeah. Maybe do a test with a single file. You can then verify if it is compressed with

```
mc stat --debug play/testbucket/0.txt
....
mc: <DEBUG> HEAD /testbucket/0.txt HTTP/1.1
Host: play.min.io
User-Agent: MinIO (windows; amd64) minio-go/v7.0.55 mc/DEVELOPMENT.GOGET
Authorization: AWS4-HMAC-SHA256 Credential=**REDACTED**/20230808/us-east-1/s3/aws4_request, SignedHeaders=host;x-amz-content-sha256;x-amz-date, Signature=**REDACTED**
X-Amz-Content-Sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
X-Amz-Date: 20230808T134633Z

mc: <DEBUG> HTTP/1.1 200 OK
Content-Length: 612018
Accept-Ranges: bytes
Connection: keep-alive
Content-Type: text/plain
Date: Tue, 08 Aug 2023 13:46:34 GMT
Etag: "3c83bc58e3ee52e5340b94a03b5a601b"
Last-Modified: Tue, 08 Aug 2023 13:46:05 GMT
Server: nginx/1.18.0 (Ubuntu)
Strict-Transport-Security: max-age=31536000; includeSubDomains
Vary: Origin
Vary: Accept-Encoding
X-Amz-Bucket-Region: us-east-1
X-Amz-Id-2: 3e996b2f640d7e065d3a5c4e39a5538cefb82e3e77771990265e4698d8681eac
X-Amz-Request-Id: 17796C772B9986CF
X-Content-Type-Options: nosniff
X-Minio-Compressed: s2
X-Xss-Protection: 1; mode=block
```

`X-Minio-Compressed: s2` indicates it is compressed.

**Reply by leepfrog-ger** [reply]
Has there been any development in this regard or is the best way still to copy/move the files to apply compression?

**Reply by klauspost** [reply]
Still no timeframe. We do not have much customer interest at this point, so it remains a low priority feature.

**Reply by leepfrog-ger** [reply]
Thanks for the update!

For everyone else interested in this: we've validated that a `mc mv` into another bucket indeed does apply the compression.
